web/app/js/components/TapEventTable.jsx

```diff
@@ -61,149 +85,86 @@ let tapColumns = (resourceType, filterOptions, ResourceLink) => [
-      },
-      {
-        title: "Path",
-        key: "path",
-        dataIndex: "requestInit.http.requestInit",
-        filters: genFilterOptionList(filterOptions.path),
-        onFilter: (value, row) =>
-          _.get(row, "requestInit.http.requestInit.path") === value,
-        render: d => !d ? <Icon type="loading" /> : d.path
-      },
-      {
-        title: "Scheme",
-        key: "scheme",
-        dataIndex: "requestInit.http.requestInit",
-        filters: genFilterOptionList(filterOptions.scheme),
-        onFilter: (value, row) =>
-          _.get(row, "requestInit.http.requestInit.scheme.registered") === value,
-        render: d => !d ? <Icon type="loading" /> : _.get(d, "scheme.registered")
-      },
-      {
-        title: "Method",
-        key: "method",
-        dataIndex: "requestInit.http.requestInit",
-        filters: _.map(filterOptions.httpMethod, d => {
-          return { text: d, value: d};
-        }),
-        onFilter: (value, row) =>
-          _.get(row, "requestInit.http.requestInit.method.registered") === value,
-        render: d => !d ? <Icon type="loading" /> : _.get(d, "method.registered")
-      }
-    ]
-  },
-  {
-    title: "Response Init",
-    children: [
-      {
-        title: "HTTP status",
-        key: "http-status",
-        dataIndex: "responseInit.http.responseInit",
-        filters: genFilterOptionList(filterOptions.httpStatus),
-        onFilter: (value, row) =>
-          _.get(row, "responseInit.http.responseInit.httpStatus") + "" === value,
-        render: d => !d ? <Icon type="loading" /> : d.httpStatus
-      },
-      {
-        title: "Latency",
-        key: "rsp-latency",
-        dataIndex: "responseInit.http.responseInit",
-        render: d => !d ? <Icon type="loading" /> : formatTapLatency(d.sinceRequestInit)
-      },
-    ]
-  },
-  {
-    title: "Response End",
-    children: [
-      {
-        title: "GRPC status",
-        key: "grpc-status",
-        dataIndex: "responseEnd.http.responseEnd",
-        filters: grpcStatusCodeFilters,
-        onFilter: (value, row) =>
-          (_.get(row, "responseEnd.http.responseEnd.eos.grpcStatusCode") + "") === value,
-        render: d => !d ? <Icon type="loading" /> : _.get(d, "eos.grpcStatusCode")
-      },
-      {
-        title: "Latency",
-        key: "end-latency",
-        dataIndex: "responseEnd.http.responseEnd",
-        render: d => !d ? <Icon type="loading" /> : formatTapLatency(d.sinceResponseInit)
-      },
-      {
-        title: "Response Length (B)",
-        key: "rsp-length",
-        dataIndex: "responseEnd.http.responseEnd",
-        render: d => !d ? <Icon type="loading" /> : formatWithComma(d.responseBytes)
-      },
-    ]
-
   }
 ];
 
+const tapColumns = (resourceType, filterOptions, ResourceLink) => {
+  return _.concat(
+    topLevelColumns(resourceType, filterOptions, ResourceLink),
+    [ methodCol, pathCol, httpStatusCol, grpcStatusCol ]
```

Comment:
I realize it's not in the mocks, but I really miss seeing a latency number in this row. How would you feel about:

```
[ methodCol, pathCol, responseInitLatencyCol, httpStatusCol, grpcStatusCol ]
```

Reply:
I second this, although responseEndLatency is possibly a better latency metric to show than responseInitLatency.

Reply:
@adleong As far as I can tell `responseEndLatency` measure the time between response init and response end. So for unary requests, it's almost always displayed as microseconds, and doesn't reflect the actual duration of the request. It looks like this:

![image](https://user-images.githubusercontent.com/9226/45572018-03ac8380-b81d-11e8-8ef9-590482c1dafe.png)

Reply:
Ah, you're right.  I was looking for time from request init to response end.  I guess we would have to sum requestInit<->responseInit and responseInit<->responseEnd.

Reply:
yeah, I agree that I miss the latency here too. I'll add responseInitLatency back